sherpa/ui/utils.py

```diff
@@ -973,6 +973,53 @@ def clean(self):
             'source_component': sherpa.image.ComponentSourceImage()
         }
 
+        # Reset the generator.
+        #
+        self.set_rng(None)
+
+    def get_rng(self):
+        """Return the RNG generator in use.
+
+        The return can be None, which means the legacy NumPy
+        RandomState generator is in use: see
+        https://numpy.org/doc/stable/reference/random/legacy.html
+
+        .. versionadded:: 4.16.0
+
+        See Also
+        --------
+        set_rng
+
+        """
+
+        return self._rng
+
+    def set_rng(self, rng):
+        """Set the RNG generator.
+
+        .. versionadded:: 4.16.0
+           This replaces the seed argument for certain routines and
```

Comment:
Maybe it replaces the need for the seed parameter, need to review. Seed is used for at least

- moncar method options, which we might want to keep
- resampledata stuff, which we probably want to deprecate
- anything else???

Reply:
I agree with this assessment. Keep for moncar, get rid of it for everything else.